thirdparty/onnx/CMakeLists.txt

```diff
@@ -44,20 +44,27 @@ foreach(_onnx_target onnx onnx_proto)
         $<TARGET_PROPERTY:protobuf::libprotobuf,INTERFACE_INCLUDE_DIRECTORIES>)
 endforeach()
 
-if(WIN32)
-    # from onnx==1.13.1 it requires C++17 when compiling on Windows
-    target_compile_features(onnx PRIVATE cxx_std_17)
+if(MINGW)
     # OPTIONAL is a reserved word for mingw at least
-    if(MINGW)
-        target_compile_definitions(onnx PRIVATE OPTIONAL=OPTIONAL_PLACEHOLDER)
-    endif()
+    target_compile_definitions(onnx PRIVATE OPTIONAL=OPTIONAL_PLACEHOLDER)
 endif()
 
-# WA to support old compiler version, removes unused header which may block build
+# from onnx==1.13.1 it requires C++17 when compiling on Windows, and since onnx==1.16.0 on Linux
+target_compile_features(onnx PRIVATE cxx_std_17)
+
+# WA to allow build on Debian 9
```

Comment:
we can drop this WA as on current master we have migrated to Debian 10 with newer compiler verison